src/v2/cookbook/dockerize-vuejs-app.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Dockerize Vue.js App
+type: cookbook
+order: 8
+---
+
+## Simple Example
+
+So you built your first Vue.js app using the amazing [Vue.js webpack template](https://github.com/vuejs-templates/webpack) and now you really want to show off with your colleagues by demonstrating that you can also run it in a Docker container.
+
+Let's start by creating a `Dockerfile` in the root folder of our project:
+
+```docker
+FROM node:9.11.1-alpine
+
+# install simple http server for serving static content
+RUN npm install -g http-server
+
+# make the 'app' folder the current working directory
+WORKDIR /app
+
+# copy both 'package.json' and 'package-lock.json' (if available)
+COPY package*.json ./
+
+# install project dependencies
+RUN npm install
+
+# copy project files and folders to the current working directory (i.e. 'app' folder)
+COPY . .
+
+# build app for production with minification
+RUN npm run build
+
+EXPOSE 8080
+CMD [ "http-server", "dist" ]
+```
+
+It may seem reduntant to first copy `package.json` and `package-lock.json` and then all project files and folders in two separate steps but there is actually [a very good reason for that](http://bitjudo.com/blog/2014/03/13/building-efficient-dockerfiles-node-dot-js/) (spoiler: it allows us to take advantage of cached Docker layers).
+
+Now let's build the Docker image of our Vue.js app:
+
+```bash
+docker build -t vuejs-cookbook/dockerize-vuejs-app .
+```
+
+Finally, let's run our Vue.js app in a Docker container:
+
+```bash
+docker run -it -p 8080:8080 --rm --name dockerize-vuejs-app-1 vuejs-cookbook/dockerize-vuejs-app
+```
+
+We should be able to access our Vue.js app on `localhost:8080`.
+
+## Real-World Example
+
+In the previous example, we used a simple, zero-configuration command-line [http server](https://github.com/indexzero/http-server) to serve our Vue.js app. Your colleagues are impressed but your boss is still shaking his head because we chose not to stand on the shoulders of some giant like [NGINX](https://www.nginx.com/) or [Apache](https://httpd.apache.org/). Now we are going to make it right.
```

Comment:
This paragraph reads pretty strangely to me- it's not terribly important to be impressive to your colleagues, (not to mention the pronoun should be 'their' for boss), but there' not an explanation of _why_ it's helpful to use nginx here, @henripal hinted at this earlier. I'm not questioning your use of nginx, but there should be some technical explanation of why we might make a change like this, for clarity.

Reply:
@sdras, as usual, thanks for your feedback.

I guess the whole point of the fictionary scene was to point out that the reader's boss was not impressed by the work we (me and the reader) did in the simple example because he (the boss) already knew about "better" ways to serve the app (e.g. NGINX and Apache)! 😅 

Anyway, I've updated the paragraph to be a little bit more formal and explicit, especially in explaining why I've decided to use NGINX for the real-world example.

Please do let me know if this new version makes sense to you.